mysite/ct/tests.py

```diff
@@ -291,3 +297,15 @@ def test_refresh_timer(self):
         s = pageData.get_refresh_timer(request)
         self.assertNotEqual(s, '0:00')
         self.assertEqual(s[:3], '0:0')
+
+
+class SMTPerrorTest(TestCase):
+
+    @patch('psa.mail.send_mail')
+    def smtp_error(self, mocked):
+        mocked.side_effect = smtplib.SMTPServerDisconnected
+        post_data = {'csrfmiddlewaretoken': _get_new_csrf_key(),
```

Comment:
`csrfmiddlewaretoken` is not required for python social auth complete view:

    @never_cache
    @csrf_exempt
    @psa('{0}:complete'.format(NAMESPACE))
    def complete(request, backend, *args, **kwargs):
        """Authentication complete view"""
        return do_complete(request.backend, _do_login, request.user,
                           redirect_name=REDIRECT_FIELD_NAME, *args, **kwargs)